roiextractors/extractors/numpyextractors/numpyextractors.py

```diff
@@ -222,47 +208,31 @@ def roi_locs(self):
         else:
             return self._roi_locs
 
-    @property
-    def num_of_frames(self):
-        return self.roi_response.shape[1]
-
-    @property
-    def samp_freq(self):
-        if self._samp_freq is None:
-            time = self.total_time
-            nframes = self.num_of_frames
-            try:
-                return nframes / time
-            except ZeroDivisionError:
-                return 0
-        else:
-            return self._samp_freq
-
     @staticmethod
     def write_segmentation(segmentation_object, savepath):
         raise NotImplementedError
 
     # defining the abstract class enformed methods:
     def get_roi_ids(self):
-        return list(range(self.no_rois))
+        if self._roi_ids is None:
+            return list(range(self.no_rois))
+        else:
+            return self._roi_ids
 
     def get_num_rois(self):
-        return self.no_rois
+        return self.image_masks.shape[2]
 
     def get_roi_locations(self, roi_ids=None):
         if roi_ids is None:
-            return self.roi_locs
+            return self.roi_locations
         else:
-            roi_idx = [np.where(np.array(i) == self.roi_idx)[0] for i in roi_ids]
+            roi_idx = [np.where(np.array(i) == self.roi_ids)[0] for i in roi_ids]
             ele = [i for i, j in enumerate(roi_idx) if j.size == 0]
             roi_idx_ = [j[0] for i, j in enumerate(roi_idx) if i not in ele]
-            return self.roi_locs[:, roi_idx_]
+            return self.roi_locations[:, roi_idx_]
 
     def get_num_frames(self):
-        return self.num_of_frames
-
-    def get_sampling_frequency(self):
-        return self.samp_freq
+        return self._roi_response.shape[1]
```

Comment:
Is this the sampling frequency?

Reply:
yes, this is the samp rate of the traces.

Reply:
maybe this is the length of the response, i.e. the number of samples?

Reply:
```
def get_num_frames(self):
      return self._roi_response.shape[1]
```
this refers to the number of frames. 
`get_sampling_frequency()` method is in the parent class

Reply:
oh, my fault, I was not reading the line changes correctly

Reply:
When I open the file mentioned here, it does not show what you're mentioning lol
This is what I see:
![image](https://user-images.githubusercontent.com/11567561/90428721-090a4c80-e08a-11ea-936e-4835ac96398a.png)